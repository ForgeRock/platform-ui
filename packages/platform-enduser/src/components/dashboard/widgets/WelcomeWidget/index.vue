<template>
  <BJumbotron class="text-center">
    <template slot="header">
      <BImg
        :src="require('@/assets/images/profile-default.png')"
        rounded="circle"
        width="112"
        height="112"
        alt="img"
        class="m-1 mb-3" />
      <div>{{ $t('pages.dashboard.widgets.welcome.greeting') }}, {{ fullName }}</div>
    </template>
    <template slot="lead">
      <div>
        {{ $t('pages.dashboard.widgets.welcome.welcomeMessage') }}
      </div>
      <BButton
        @click="openProfile()"
        variant="primary"
        class="mt-2">
        {{ $t('pages.dashboard.widgets.welcome.editProfile') }}
      </BButton>
    </template>
  </BJumbotron>
</template>

<script>
import { startCase } from 'lodash';
import { mapState } from 'vuex';

/**
 * @description Widget that provides a welcome message for the managed resource, also provides a button to directly access editing the resources profile.
 *
 * */
export default {
  name: 'WelcomeWidget',
  props: {
    userDetails: {
      type: Object,
      default: () => {},
    },
    widgetDetails: {
      type: Object,
      default: () => {},
    },
  },
  data() {
    return {};
  },
  mounted() {},
  methods: {
    openProfile() {
      this.$router.push({ name: 'Profile', params: { openProfile: !this.internalUser } });
    },
  },
  computed: {
    ...mapState({
      internalUser: (state) => state.UserStore.internalUser,
    }),
    fullName() {
      let fullName = '';

      if (this.userDetails.givenName.length > 0 || this.userDetails.sn.length > 0) {
        fullName = startCase(`${this.userDetails.givenName} ${this.userDetails.sn}`);
      } else {
        fullName = this.userDetails.userId;
      }

      return fullName;
    },
  },
};
</script>

<style lang="scss" scoped></style>
