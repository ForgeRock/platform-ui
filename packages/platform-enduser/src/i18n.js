/**
 * Copyright (c) 2020-2024 ForgeRock. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

import { createI18n } from 'vue-i18n';
import { merge } from 'lodash';
import dateTimeFormats from '@forgerock/platform-shared/src/dateTimeFormats.json';

function loadLocaleMessages() {
  const locales = require.context('./locales', true, /[A-Za-z0-9-_,\s]+\.json$/i);
  // Can't use workspace symlink due to jest issues
  const sharedLocales = require.context('../../platform-shared/src/locales', true, /[A-Za-z0-9-_,\s]+\.json$/i);
  const messages = {};

  locales.keys().forEach((key) => {
    const matched = key.match(/([A-Za-z0-9-_]+)\./i);
    if (matched && matched.length > 1) {
      const locale = matched[1];
      let generatedTranslation = locales(key);

      // Check for shared local and merge in if it exists
      const sharedLocaleExists = sharedLocales.keys().includes(key);
      if (sharedLocaleExists) {
        generatedTranslation = merge(sharedLocales(key), generatedTranslation);
      }

      messages[locale] = generatedTranslation;
    }
  });

  return messages;
}

export default createI18n({
  messages: loadLocaleMessages(),
  silentFallbackWarn: true,
  silentTranslationWarn: true,
  dateTimeFormats,
});
