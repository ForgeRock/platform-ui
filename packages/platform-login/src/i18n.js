/**
 * Copyright (c) 2020-2021 ForgeRock. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

import Vue from 'vue';
import VueI18n from 'vue-i18n';
import { merge } from 'lodash';
import dateTimeFormats from '@forgerock/platform-shared/src/dateTimeFormats.json';

Vue.use(VueI18n);

function loadLocaleMessages() {
  const locales = require.context('./locales', true, /[A-Za-z0-9-_,\s]+\.json$/i);
  // Can't use workspace symlink due to jest issues
  const sharedLocales = require.context('../../platform-shared/src/locales', true, /[A-Za-z0-9-_,\s]+\.json$/i);
  const messages = {};

  locales.keys().forEach((key) => {
    const matched = key.match(/([A-Za-z0-9-_]+)\./i);
    if (matched && matched.length > 1) {
      const locale = matched[1];
      let generatedTranslation = locales(key);

      // Check for shared local and merge in if it exists
      const sharedLocaleExists = sharedLocales.keys().includes(key);
      if (sharedLocaleExists) {
        generatedTranslation = merge(sharedLocales(key), generatedTranslation);
      }

      messages[locale] = generatedTranslation;
    }
  });

  return messages;
}

Vue.mixin({
  methods: {
    translationExists(path) {
      return this.$t(path) !== path;
    },
  },
});

export default new VueI18n({
  locale: process.env.VUE_APP_I18N_LOCALE || 'en',
  fallbackLocale: process.env.VUE_APP_I18N_FALLBACK_LOCALE || 'en',
  messages: loadLocaleMessages(),
  silentTranslationWarn: true,
  dateTimeFormats,
});
