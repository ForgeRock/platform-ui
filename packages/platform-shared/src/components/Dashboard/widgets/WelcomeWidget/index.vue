<!-- Copyright (c) 2020-2023 ForgeRock. All rights reserved.

This software may be modified and distributed under the terms
of the MIT license. See the LICENSE file for details. -->
<template>
  <div
    v-if="displayCompactHeader"
    class="container my-4"
    data-testid="compactHeader">
    <BMedia>
      <template #aside>
        <BAvatar
          variant="link"
          size="72px"
          :src="avatarSource" />
      </template>
      <h1 class="text-capatilize">
        {{ $t(`pages.dashboard.widgets.welcome.greeting${timeOfDay}`, { name: givenName }) }}
      </h1>
      <p>{{ $t('pages.dashboard.widgets.welcome.welcomeMessageCompact') }}</p>
    </BMedia>
  </div>
  <div
    v-else
    data-testid="defaultHeader"
  >
    <BJumbotron class="text-center">
      <template #header>
        <BAvatar
          variant="link"
          size="112px"
          :src="avatarSource" />
        <div data-testid="dashboard-welcome-greeting">
          {{ $t('pages.dashboard.widgets.welcome.greeting') }}, <span class="text-capitalize">
            {{ name }}
          </span>
        </div>
      </template>
      <template #lead>
        <div v-if="!$store.state.isFraas">
          {{ $t('pages.dashboard.widgets.welcome.welcomeMessage') }}
        </div>
        <BButton
          @click="openProfile()"
          variant="primary"
          class="mt-2">
          {{ $t('pages.dashboard.widgets.welcome.editProfile') }}
        </BButton>
      </template>
    </BJumbotron>
  </div>
</template>

<script>
import {
  BAvatar,
  BButton,
  BJumbotron,
  BMedia,
} from 'bootstrap-vue';
import { mapState } from 'pinia';
import { useUserStore } from '@forgerock/platform-shared/src/stores/user';
import { useEnduserStore } from '@forgerock/platform-shared/src/stores/enduser';

/**
 * @description Widget that provides a welcome message for the managed resource, also provides a button to directly access editing the resources profile.
 */
export default {
  name: 'WelcomeWidget',
  props: {
    widgetDetails: {
      type: Object,
      default: () => {},
    },
    displayCompactHeader: {
      type: Boolean,
      default: false,
    },
  },
  components: {
    BAvatar,
    BButton,
    BJumbotron,
    BMedia,
  },
  data() {
    return {
      timeOfDay: '',
    };
  },
  mounted() {
    this.getTimeOfDay(new Date().getHours());
  },
  methods: {
    openProfile() {
      this.$router.push({ name: 'Profile' });
    },
    getTimeOfDay(currentHour) {
      if (currentHour >= 0 && currentHour < 12) {
        this.timeOfDay = 'Morning';
      } else if (currentHour >= 12 && currentHour < 18) {
        this.timeOfDay = 'Afternoon';
      } else {
        this.timeOfDay = 'Evening';
      }
    },
  },
  computed: {
    ...mapState(useUserStore, ['name', 'givenName']),
    ...mapState(useEnduserStore, ['profileImage']),
    avatarSource() {
      return this.profileImage.length > 0 ? this.profileImage : require('@forgerock/platform-shared/src/assets/images/avatar.png'); // eslint-disable-line global-require
    },
  },
};
</script>
<style lang=scss scoped>
@include media-breakpoint-down(xs) {
  .media {
    flex-direction: column;
  }
}
</style>
