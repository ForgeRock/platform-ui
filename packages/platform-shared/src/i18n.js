/**
 * Copyright 2019 ForgeRock AS. All Rights Reserved
 *
 * Use of this code requires a commercial software license with ForgeRock AS.
 * or with one of its affiliates. All use shall be exclusively subject
 * to such license between the licensee and ForgeRock AS.
 */
import Vue from 'vue';
import VueI18n from 'vue-i18n';
import dateTimeFormats from './dateTimeFormats.json';

Vue.use(VueI18n);

function loadLocaleMessages() {
  const locales = require.context('./locales', true, /[A-Za-z0-9-_,\s]+\.json$/i);
  const messages = {};

  locales.keys().forEach((key) => {
    const matched = key.match(/([A-Za-z0-9-_]+)\./i);
    if (matched && matched.length > 1) {
      const locale = matched[1];
      messages[locale] = locales(key);
    }
  });

  return messages;
}

export default new VueI18n({
  locale: process.env.VUE_APP_I18N_LOCALE || 'en',
  fallbackLocale: process.env.VUE_APP_I18N_FALLBACK_LOCALE || 'en',
  messages: loadLocaleMessages(),
  silentTranslationWarn: true,
  dateTimeFormats,
});
