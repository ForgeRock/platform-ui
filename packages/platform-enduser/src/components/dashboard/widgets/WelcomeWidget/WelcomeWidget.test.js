/**
 * Copyright (c) 2020-2021 ForgeRock. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

import BootstrapVue from 'bootstrap-vue';
import { createLocalVue, shallowMount } from '@vue/test-utils';
import i18n from '@/i18n';
import WelcomeWidget from '@/components/dashboard/widgets/WelcomeWidget';

const localVue = createLocalVue();
localVue.use(BootstrapVue);

describe('Dashboard.vue', () => {
  it('Welcome widget loaded', () => {
    const wrapper = shallowMount(WelcomeWidget, {
      localVue,
      i18n,
      propsData: {
        userDetails: {
          givenName: 'test',
          sn: 'test',
        },
      },
    });

    expect(wrapper.name()).toBe('WelcomeWidget');
  });

  it('renders default header if displayCompactHeader is set to false', () => {
    const wrapper = shallowMount(WelcomeWidget, {
      localVue,
      i18n,
      propsData: {
        displayCompactHeader: false,
      },
    });
    expect(wrapper.find('[data-testid="defaultHeader"]').isVisible()).toBe(true);
  });

  it('renders compact header if displayCompactHeader is set to true', () => {
    const wrapper = shallowMount(WelcomeWidget, {
      localVue,
      i18n,
      propsData: {
        userDetails: {
          givenName: 'Test',
        },
        displayCompactHeader: true,
      },
    });
    expect(wrapper.find('[data-testid="compactHeader"]').isVisible()).toBe(true);
  });

  it('greets user according to time of day and given name', () => {
    WelcomeWidget.mounted = jest.fn();
    const wrapper = shallowMount(WelcomeWidget, {
      localVue,
      i18n,
      propsData: {
        userDetails: {
          givenName: 'Test',
        },
        displayCompactHeader: true,
      },
      data() {
        return {
          timeOfDay: 'Morning',
        };
      },
    });
    expect(wrapper.find('h1').text()).toBe('Good Morning Test!');
  });

  it('sets greeting timeOfDay of compact header to: Morning, from hour 0 to 11', () => {
    const wrapper = shallowMount(WelcomeWidget, {
      localVue,
      i18n,
      propsData: {
        userDetails: {
          givenName: 'Test',
        },
        displayCompactHeader: true,
      },
      data() {
        return {
          timeOfDay: '',
        };
      },
    });
    const currentHour = 11;
    wrapper.vm.getTimeOfDay(currentHour);

    expect(wrapper.vm.timeOfDay).toBe('Morning');
  });

  it('sets greeting timeOfDay of compact header to: Afternoon, from hour 12 up until, and icnluding, hour 17', () => {
    const wrapper = shallowMount(WelcomeWidget, {
      localVue,
      i18n,
      propsData: {
        userDetails: {
          givenName: 'Test',
        },
        displayCompactHeader: true,
      },
      data() {
        return {
          timeOfDay: '',
        };
      },
    });
    const currentHour = 12;
    wrapper.vm.getTimeOfDay(currentHour);

    expect(wrapper.vm.timeOfDay).toBe('Afternoon');
  });

  it('sets greeting timeOfDay of compact header to: Evening, from hour 18 up until, and icnluding, hour 23', () => {
    const wrapper = shallowMount(WelcomeWidget, {
      localVue,
      i18n,
      propsData: {
        userDetails: {
          givenName: 'Test',
        },
        displayCompactHeader: true,
      },
      data() {
        return {
          timeOfDay: '',
        };
      },
    });
    const currentHour = 18;
    wrapper.vm.getTimeOfDay(currentHour);

    expect(wrapper.vm.timeOfDay).toBe('Evening');
  });
});
