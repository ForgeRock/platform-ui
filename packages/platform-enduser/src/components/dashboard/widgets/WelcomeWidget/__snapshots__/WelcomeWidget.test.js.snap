// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard.vue Welcome widget loaded 1`] = `<bjumbotron-stub headertag="h1" headerlevel="3" leadtag="p" tag="div" class="text-center"></bjumbotron-stub>`;
